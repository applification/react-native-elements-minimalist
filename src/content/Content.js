import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

class Content extends Component {
  static propTypes = {
    backgroundColor: PropTypes.string.isRequired,
  };

  static defaultProps = {
    backgroundColor: '#FFFFFF',
  };

  render() {
    return (
      <View style={{ flex: 1, backgroundColor: this.props.backgroundColor }}>
        {this.props.children}
      </View>
    );
  }
}

export default Content;
