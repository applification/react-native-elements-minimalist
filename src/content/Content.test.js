import React from 'react';
import renderer from 'react-test-renderer';

import Content from './Content';

describe('CONTENT VIEW', () => {
  it('renders a Content Component using defaultProps of white background', () => {
    expect(renderer.create(<Content />)).toMatchSnapshot();
  });

  it('renders a Content Component using passed in prop of cyan background', () => {
    expect(
      renderer.create(<Content backgroundColor="cyan" />)
    ).toMatchSnapshot();
  });

  // TEST for iOS paddingTop 30
});
