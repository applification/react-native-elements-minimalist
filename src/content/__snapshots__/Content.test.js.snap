// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CONTENT VIEW renders a Content Component using defaultProps of white background 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
/>
`;

exports[`CONTENT VIEW renders a Content Component using passed in prop of cyan background 1`] = `
<View
  style={
    Object {
      "backgroundColor": "cyan",
      "flex": 1,
    }
  }
/>
`;
